### DISCUSSION TITLE
Recurring transaction didn't register on expected date

### DISCUSSION THREAD

**Discussion by bradgy** [firefly-iii/firefly-iii#4825] [Firefly III]
I am a new user so I am fully expecting this to be a problem between computer and chair, with that said;

I set up a couple of recurring transactions a few days ago to run today (2nd June), which didn't seem to work properly. That is, the transactions I expected to see in the relevant expenses/transfers/deposits pages weren't there.

I run firefly-iii in docker, and used the host cron to run the job with this command, as per the docs:

`0 3 * * * docker exec --user www-data <my-own-firefly-container-name> /usr/local/bin/php /var/www/html/artisan firefly-iii:cron`

Before today, the "next due" dates on the recurring transactions page looked correct: June 2, 2021 and June 16, 2021 (it's a fortnightly/every-2nd-week recurrence). Today, the dates have updated to June 16, 2021, and June 30, 2021. As an aside, I ran into the same recurring transaction issue as #4798, which I worked around by starting the recurring transaction the wednesday before (26/05/2021) I wanted it to start (02/06/2021).

Interestingly, the auto-budget items I have set up, worked exactly as expected when the cronjob fired at the 1st of the month. So I assume the cron is set up correctly.

Would appreciate any help or ideas - especially wondering if perhaps I haven't set something up correctly.

**Comment by bradgy** [comment]
Thinking now this issue might be a bug with the 'next due' readout?

My recurring transactions fired overnight, despite me not expecting them til the following week as per the next due dates.

To check, I deleted the existing recurring transactions and reentered them. Both starting 16th June and skipping every one week.

'Next Due' reports the alternate weeks, which are the ones I want skipped.

![Screenshot from 2021-06-09 10-33-00](https://user-images.githubusercontent.com/5727821/121274833-ad59ca00-c90e-11eb-9138-1aa807b9841a.png)

![Screenshot from 2021-06-09 10-37-44](https://user-images.githubusercontent.com/5727821/121274924-e42fe000-c90e-11eb-9dce-617d9bbfc4bf.png)

![Screenshot from 2021-06-09 10-33-26](https://user-images.githubusercontent.com/5727821/121274942-eabe5780-c90e-11eb-824c-d46d615ff835.png)

Hope this makes things a bit clearer.

----------------------------------

Debug information generated at 2021-06-09 10:42:09 Australia/Melbourne for Firefly III version **5.5.12**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.5.12 |
| Firefly III API | 1.5.2 |
| PHP | 8.0.6 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | e16490ca-486d-4a0f-809f-1b90416a5b61 |
| Using docker? | true |
| Telemetry | false |
| Layout |  |
| System TZ | Australia/Melbourne |
| Browser TZ | Australia/Melbourne |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| DB version | 16 (exp. 16) |
| Login provider | eloquent  |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| Session start | 2021-06-01 00:00:00 |
| Session end | 2021-06-30 00:00:00 |
| Session first | 2021-05-29 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:89.0) Gecko/20100101 Firefox/89.0 |

**Comment by JC5** [comment]
I'm not sure how I missed this one, sorry about that. I'll put it on my list and get back to you!

**Comment by JC5** [comment]
The issue from #4798 is related to this issue, and will be fixed in the next release :)

**Reply by bradgy** [reply]
I went back and read #4798 properly, and I agree that it's exactly the problem I was experiencing. Appreciate you looking into it! 

Fwiw, the transactions I was expecting to fire overnight did so (despite the "next due" saying it wouldn't happen until next week).
